app/main.py

```diff
@@ -1 +1,54 @@
-#  write your code here
+class Animal:
+    animals = []
+
+    def __init__(self, name: str, appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> str:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, appetite: int = 3,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+        Animal.animals.append(self)
+
+    @staticmethod
+    def catch_mouse() -> str:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, appetite: int = 7,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+        Animal.animals.append(self)
+
+    @staticmethod
+    def bring_slippers() -> str:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    points = 0
+    for animal in animals:
+        if animal.is_hungry:
+            animal.feed()
+            points += animal.appetite
```

Comment:
U may do it much shorter

Reply:
can't make it with list comprehension

Reply:
try to use sum function here

Reply:
can't call animal.feed() and sum(animal appetite) simultaniously in a list comprehension

Reply:
animal.feed returns animal appetite, so u can iterate through animals and call feed method for every animal and pass this generator to sum function